### PR TITLE
feat: Changed OCI tag format

### PR DESCRIPTION
**Pull request by stevehipwell**
Fixes #N/A 

**Description**
This PR changes the OCI tag pattern to be the SemVer version without the `v` prefix.

**How was this change tested?**

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: hack/release/common.sh [hunk 1]
```diff
@@ -66,136 +53,165 @@ authenticatePrivateRepo() {
...
-  for VERSION in "${VERSIONS[@]}"; do
-    yq -i ".params.versions += \"${VERSION}\"" website/hugo.yaml
+  for version in "${versions[@]}"; do
+    yq -i ".params.versions += \"${version}\"" website/hugo.yaml
   done
 }
+
+# updateKarpenterCoreGoMod bumps the karpenter-core go.mod to the release version so that the
+# karpenter and karpenter-core release versions match
+updateKarpenterCoreGoMod(){
```
Comment (jonathan-innis) [inline hack/release/common.sh:206]: I think we can drop this function from the stable PR. We ensure we pin the version through a helm update ahead-of-time so there's no need for this function to exist right now AFAIK

File: hack/release/common.sh [hunk 2]
```diff
@@ -66,136 +53,165 @@ authenticatePrivateRepo() {
...
-    rm "${HELM_CHART_FILE_NAME}"
-    cd ..
-
-    cosignHelmChart "${RELEASE_REPO}/${CHART_NAME}" "${HELM_CHART_VERSION}"
+buildDate() {
+  local date_epoch
+
+  date_epoch="${1}"
+
+  date -u -r "${date_epoch}" "+%Y-%m-%dT%H:%M:%SZ" 2>/dev/null
```
Comment (jonathan-innis) [inline hack/release/common.sh:129]: What's the reason/need to change the command that is being called to produce the buildDate from before?

File: hack/release/stable-pr.sh [hunk 3]
```diff
@@ -1,42 +1,25 @@
...
-if [[ $(releaseType $GIT_TAG) != $RELEASE_TYPE_STABLE ]]; then
+git_tag="$(git describe --exact-match --tags || echo "none")"
+if [[ "${git_tag}" != v* ]]; then
   echo "Not a stable release. Missing required git tag."
   exit 1
 fi
 
-versionData "$GIT_TAG"
-updateKarpenterCoreGoMod "$GIT_TAG"
+updateKarpenterCoreGoMod "${git_tag}"
```
Comment (jonathan-innis) [inline hack/release/stable-pr.sh:14]: I think we should be able to drop this command now

File: hack/release/release.sh [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
-HEAD_HASH=$(git rev-parse HEAD)
-
-SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
+SCRIPT_DIR="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)"
+# shellcheck source=hack/release/common.sh
 source "${SCRIPT_DIR}/common.sh"
-config
+
+git_tag="$(git describe --exact-match --tags || echo "no tag")"
+if [[ "${git_tag}" == "no tag" ]]; then
```
Comment (jonathan-innis) [inline hack/release/release.sh:9]: What's the reason this conditional check is different from the one that we use for `release-crd.sh`?

File: hack/release/common.sh [hunk 5]
```diff
@@ -66,136 +53,165 @@ authenticatePrivateRepo() {
...
-    SOURCE_DATE_EPOCH=$(git log -1 --format='%ct')
-    echo "$(date -u -r "${SOURCE_DATE_EPOCH}" $DATE_FMT 2>/dev/null)"
+cosignOciArtifact() {
+  local commit_sha build_date artifact
+
+  commit_sha="${1}"
+  build_date="${2}"
+  artifact="${3}"
+
+  cosign sign --yes -a commitSha="${commit_sha}" -a buildDate="${build_date}" "${artifact}"
```
Comment (jonathan-innis) [inline hack/release/common.sh:117]: Seems like we aren't attesting the GIT_VERSION anymore?

File: hack/release/common.sh [hunk 6]
```diff
@@ -66,136 +53,165 @@ authenticatePrivateRepo() {
...
-    RELEASE_REPO=$1
-    HELM_CHART_VERSION=$2
-    digest="$(crane digest "${RELEASE_REPO}:${HELM_CHART_VERSION}")"
-    cosign sign --yes "${RELEASE_REPO}:${HELM_CHART_VERSION}@${digest}"
+prepareWebsite() {
+  local version version_parts short_version
+
+  version="${1}"
+  # shellcheck disable=SC2206
+  version_parts=(${version//./ })
```
Comment (jonathan-innis) [inline hack/release/common.sh:137]: I'm not a bash expert, can you point me to docs that explain what this does?

File: hack/release/common.sh [hunk 7]
```diff
@@ -66,136 +53,165 @@ authenticatePrivateRepo() {
...
+  mkdir -p "website/content/en/v${short_version}"
+  cp -r website/content/en/preview/* "website/content/en/v${short_version}/"
 
-    # Update parameterized variables in the preview documentation to be statically set in the versioned documentation
-    find "website/content/en/${RELEASE_MINOR_VERSION}/" -type f | xargs perl -i -p -e "s/{{< param \"latest_release_version\" >}}/${RELEASE_VERSION}/g;"
-    find "website/content/en/${RELEASE_MINOR_VERSION}/" -type f | xargs perl -i -p -e "s/{{< param \"latest_k8s_version\" >}}/$(yq .params.latest_k8s_version website/hugo.yaml)/g;"
-    find website/content/en/${RELEASE_MINOR_VERSION}/*/*/*.yaml -type f | xargs perl -i -p -e "s/preview/${RELEASE_MINOR_VERSION}/g;"
-    find "website/content/en/${RELEASE_MINOR_VERSION}/" -type f | xargs perl -i -p -e "s/{{< githubRelRef >}}/\/${RELEASE_VERSION}\//g;"
+  # Update parameterized variables in the preview documentation to be statically set in the versioned documentation
+  find "website/content/en/v${short_version}/" -type f -print0 | xargs perl -i -p -e "s/{{< param \"latest_release_version\" >}}/${RELEASE_VERSION}/g;"
```
Comment (jonathan-innis) [inline hack/release/common.sh:153]: I had recalled doing this and this messed with the output format in a way that `xargs` couldn't understand. I'll double-check as well, but you're sure that adding `-print0` here doesn't mess with the success of this command?

**Review by jonathan-innis — COMMENTED** [review]
Nice work 🎉

File: hack/release/common.sh (hunk 5 above)
Comment (stevehipwell) [reply hack/release/common.sh:117]: The git version was being added as an annotation; annotations aren't checked as part of validation unless opted in to. I can add a `version` attribute which can be checked as part of verification if there is the need to correlate the SHA with a version, although I'm not sure that's necessary.

File: hack/release/common.sh (hunk 2 above)
Comment (stevehipwell) [reply hack/release/common.sh:129]: The command hasn't changed but I've refactored it to remove the unnecessary `echo` and extract the epoch as the same command was being used in multiple places.

File: hack/release/common.sh (hunk 6 above)
Comment (stevehipwell) [reply hack/release/common.sh:137]: The outer `(` & `)` make an array with the contents. The `//./` replacement syntax swaps `.` for ` ` and by not wrapping the variable in `"` it is treated as multiple words so each part of the version is an array entry.

File: hack/release/common.sh (hunk 7 above)
Comment (stevehipwell) [reply hack/release/common.sh:153]: Shellcheck is adamant that it's needed but I'll check again.

File: hack/release/common.sh (hunk 1 above)
Comment (stevehipwell) [reply hack/release/common.sh:206]: Will do.

File: hack/release/release.sh (hunk 4 above)
Comment (stevehipwell) [reply hack/release/release.sh:9]: This is guard clause and the other is a conditional extra function. I didn't modify these checks, just refactored the bash style.

File: hack/release/stable-pr.sh (hunk 3 above)
Comment (stevehipwell) [reply hack/release/stable-pr.sh:14]: Will do.

File: hack/release/common.sh (hunk 5 above)
Comment (stevehipwell) [reply hack/release/common.sh]: I've added the `version` attribute.

File: hack/release/common.sh (hunk 6 above)
Comment (jonathan-innis) [reply hack/release/common.sh:138]: That's cool and fancy, didn't know that that would work

File: hack/release/common.sh (hunk 7 above)
Comment (jonathan-innis) [reply hack/release/common.sh:154]: Sure thing. Let me know. I had also gotten warnings for not adding this before but I recalled that adding it had caused issues

Comment (stevehipwell) [reply hack/release/common.sh]: You're absolutely right, so I've supressed the Shellcheck warnings so it's clear this is the desired outcome.

**Comment by stevehipwell** [comment]
@jonathan-innis I've updated the docs and have standardised the pattern used to refer to Karpenter versions for consistency (see below).

```markdown
Karpenter version: `0.34.1`
Karpenter minimum version: `0.34.0`+
Karpenter latest patch: `0.34.x` (this shouldn't be used where the minimum version would also work)
```

**Review by jonathan-innis — DISMISSED** [review]
LGTM 🚀

**Review by jonathan-innis — DISMISSED** [review]
LGTM 🚀

**Review by jmdeal — APPROVED** [review]
LGTM 🚀

**Pull request merged**